### PR TITLE
allow configuring the dev host

### PR DESCRIPTION
**Pull request by dafydd-t**
Allow overriding of `localhost` for dev host, for use cases where we're running in docker-compose

### CODE REVIEW

File: cmd/pdc/main.go [hunk 1]
```diff
@@ -142,7 +142,10 @@ func main() {
 	sshConfig.URL = gatewayURL
 
 	if mf.DevMode {
-		setDevelopmentConfig(sshConfig, pdcClientCfg)
+		if err := setDevelopmentConfig(sshConfig, pdcClientCfg); err != nil {
```
Comment (hairyhenderson) [inline cmd/pdc/main.go:145]:
```suggestion
		if err = setDevelopmentConfig(sshConfig, pdcClientCfg); err != nil {
```

File: pkg/pdc/client.go [hunk 2]
```diff
@@ -50,13 +50,18 @@ type Config struct {
...
+	// DevHost is the host that the agent will connect to.
+	DevHost string
 }
 
 func (cfg *Config) RegisterFlags(fs *flag.FlagSet) {
 	var deprecated string
 	fs.StringVar(&cfg.Token, "token", "", "The token to use to authenticate with Grafana Cloud. It must have the pdc-signing:write scope")
 	fs.StringVar(&cfg.HostedGrafanaID, "gcloud-hosted-grafana-id", "", "The ID of the Hosted Grafana instance to connect to")
 	fs.StringVar(&cfg.DevNetwork, "dev-network", "", "[DEVELOPMENT ONLY] the network the agent will connect to")
+	fs.StringVar(&cfg.DevHost, "dev-network", "localhost", "[DEVELOPMENT ONLY] the host the agent will connect to")
```
Comment (hairyhenderson) [inline pkg/pdc/client.go:64]: `dev-network` is duplicated, did you mean `dev-host`?

```suggestion
	fs.StringVar(&cfg.DevHost, "dev-host", "localhost", "[DEVELOPMENT ONLY] the host the agent will connect to")
```

Comment (dafydd-t) [reply pkg/pdc/client.go:64]: heh yes definitely!

**Comment by dafydd-t** [comment]
closing in favour of https://github.com/grafana/pdc-agent/pull/108

**Pull request closed without merging**